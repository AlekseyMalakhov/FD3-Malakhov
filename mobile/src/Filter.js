import React from 'react';
import './index.css';
import ee from "./Emitter.js";

class Filter extends React.PureComponent {
    constructor(props) {
        super(props);
    }

    render() {
        return <div>
            <button type="button" className = {(this.props.view === "view_all") ? "selected" : ""} onClick = {() => ee.emit("view_all")}>Все</button>
            <button type="button" className = {(this.props.view === "view_active") ? "selected" : ""} onClick = {() => ee.emit("view_active")}>Активные</button>
            <button type="button" className = {(this.props.view === "view_blocked") ? "selected" : ""} onClick = {() => ee.emit("view_blocked")}>Заблокированные</button>
        </div>;
    }
}

export default Filter;