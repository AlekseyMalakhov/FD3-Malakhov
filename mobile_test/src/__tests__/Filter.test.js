'use strict';
import React from 'react';
import Filter from '../Filter.js';
import ee from "../Emitter.js";
import { create, act } from "react-test-renderer";

var view = "view_all";

ee.on("view_all", () => {view = "view_all"});
ee.on("view_active", () => {view = "view_active"}); 
ee.on("view_blocked", () => {view = "view_blocked"});


test("Filter works correctly", () => {
    let component;

    act(() => {
        component = create(
          <Filter view = {view}></Filter>
        );
    });

    const instance = component.root; 

    const button_all = instance.find((node) => {return node.children[0] === "Все"});
    const button_active = instance.find((node) => {return node.children[0] === "Активные"});
    const button_blocked = instance.find((node) => {return node.children[0] === "Заблокированные"});

    act(button_active.props.onClick);
    expect(view).toBe("view_active");

    act(button_blocked.props.onClick);
    expect(view).toBe("view_blocked");

    act(button_all.props.onClick);
    expect(view).toBe("view_all");
});