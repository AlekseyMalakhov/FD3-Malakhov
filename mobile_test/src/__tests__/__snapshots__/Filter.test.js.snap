// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter works correctly 1`] = `
<div>
  <button
    className="selected"
    onClick={[Function]}
    type="button"
  >
    Все
  </button>
  <button
    className=""
    onClick={[Function]}
    type="button"
  >
    Активные
  </button>
  <button
    className=""
    onClick={[Function]}
    type="button"
  >
    Заблокированные
  </button>
</div>
`;
